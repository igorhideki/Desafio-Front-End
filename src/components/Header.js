import React from 'react';

const Header = () => (
  <div className="header">
    <span className="header__icon-menu"><i className="fas fa-bars fa-fw"></i></span>
    <span className="header__title">Comunicados</span>
    <span className="header__icon-search"><i className="fas fa-search fa-fw"></i></span>    
  </div>
);

export default Header;
