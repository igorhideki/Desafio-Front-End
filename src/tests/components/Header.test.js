import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

let wrapper;

beforeAll(() => {
  wrapper = shallow(<Header />);
});

test('should render Header correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should open menu correctly', () => {
  wrapper.find('#menu-toggler').simulate('click');
  expect(wrapper.state('isOpen')).toBe(true);
});

test('should close menu correctly when clicked in overlay', () => {
  wrapper.setState(() => ({ isOpen: true }));
  wrapper.find('#overlay').simulate('click');
  expect(wrapper.state('isOpen')).toBe(false);
});

