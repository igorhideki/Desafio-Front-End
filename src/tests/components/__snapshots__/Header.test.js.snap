// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div>
  <header
    className="header"
  >
    <span
      className="header__icon-menu"
      id="menu-toggler"
      onClick={[Function]}
    >
      <i
        className="fas fa-bars fa-fw"
      />
    </span>
    <span
      className="header__title"
    >
      Comunicados
    </span>
    <span
      className="header__icon-search"
    >
      <i
        className="fas fa-search fa-fw"
      />
    </span>
  </header>
  <nav
    className="menu"
  >
    <div
      className="menu__brand"
    >
      <span>
        Comunicados
      </span>
    </div>
    <ul
      className="menu__list"
    >
      <li
        className="menu__item"
      >
        <NavLink
          activeClassName="menu__link--active"
          ariaCurrent="true"
          className="menu__link"
          exact={true}
          onClick={[Function]}
          to="/"
        >
          <i
            className="fas fa-home fa-fw menu__icon-link"
          />
          Home
        </NavLink>
      </li>
      <li
        className="menu__item"
      >
        <NavLink
          activeClassName="menu__link--active"
          ariaCurrent="true"
          className="menu__link"
          onClick={[Function]}
          to="/about"
        >
          <i
            className="fas fa-info-circle fa-fw menu__icon-link"
          />
          About
        </NavLink>
      </li>
    </ul>
  </nav>
  <div
    className="overlay"
    id="overlay"
    onClick={[Function]}
  />
</div>
`;
